developer_docs/ansible_dev_setup.asciidoc

```diff
@@ -0,0 +1,142 @@
+
+[[foreman-plugin]]
+== Foreman plugin
+
+Download https://github.com/theforeman/foreman_ansible[foreman_ansible] plugin
+[source, bash]
+....
+git clone https://github.com/theforeman/foreman_ansible.git
+....
+
+Enable `foreman_ansible` plugin
+[source, ruby]
+....
+# bundler.d/ansible.local.rb
+
+gem 'foreman_ansible', path: '/path/to/foreman_ansible'
+....
+
+Install plugin dependencies & migrate database
+[source, bash]
+....
+bundle install
+npm install
+
+bundle exec rails db:migrate
+....
+
+
+[[smart-proxy-plugin]]
+== Smart Proxy plugin
+Add plugin gems
+
+[source, bash]
+....
+# /path/to/smart-proxy/bundler.d/ansible.local.rb
+
+gem 'smart_proxy_ansible', git: 'https://github.com/theforeman/smart_proxy_ansible.git'
+....
+
+Install gems
+[source, bash]
+....
+bundle install
+....
+
+Enable feature
+
+[source, yaml]
+....
+# config/settings.d/ansible.yml
+---
+:enabled: true
+....
+
+See https://github.com/theforeman/smart_proxy_ansible/blob/master/settings.d/ansible.yml.example[ansible.yml.example] file for full configuration.
+
+Run all services
+[source, bash]
+....
+bundle exec foreman start
+bundle exec rake dynflow:executor
+
+# From Smart Proxy folder
+bundle exec bin/smart-proxy
+# => [I] Successfully initialized 'ansible'
+....
+
+[[foreman-collection]]
+== Foreman collection
+
+`theforeman.foreman` collection contains Ansible Modules to manage Foreman and Katello installations.
+
+Install https://docs.ansible.com/ansible/latest/collections/theforeman/foreman/index.html[Foreman Collection]
+
+[source, bash]
+....
+ansible-galaxy collection install theforeman.foreman
+....
+
+Create symbolic link to folder where you have your collections
+
+[source, bash]
+....
+sudo ln -s /home/<your-account>/.ansible/collections/ansible_collections/ /usr/share/ansible/collections/ansible_collections
+....
+
+[[foreman-callback]]
+== Foreman callback
+This https://docs.ansible.com/ansible/latest/collections/theforeman/foreman/foreman_callback.html[callback] will report facts and task events to Foreman
+
+[source, bash]
+....
+# /etc/ansible/ansible.cfg
+
+[defaults]
+callbacks_enabled = foreman
+callback_plugins = /usr/share/ansible/collections/ansible_collections/theforeman/foreman/plugins/callback
+
+[callback_foreman]
+url = 'http://localhost:3000'
+....
+
+Create simple playbook `ping_playbook.yml`
+
+[source, yaml]
+....
+---
+- name: Ping Playbook
+  hosts: all
+  become: yes
+  become_user: root
+  gather_facts: true
+  tasks:
+    - name: Show host's hostname
+      debug:
+        msg: "{{ ansible_ssh_host }}"
+....
+
+And run it against the host from prerequisities
+[source, bash]
+....
+ansible-playbook -i,<domain-of-the-host-you-want-to-register> -u root -k ping_playbook.yml
+....
+
+Check that callback works and host has been created in the Foreman
+
+[[importing-roles]]
+== Importing roles
+* Go to _Infrastructure > Smart Proxies_ > Smart Proxy and refresh features
+* In _Configure > Roles_ import all roles
```

Comment:
When trying to run a playbook (or a role) from Foreman UI on some host, I got an error saying: `Error running command '{} ansible-runner run /tmp/d20220829-101624-1qt623b -p playbook.yml': RuntimeError - No such file or directory - ansible-runner`
I'm using Fedora 35, and I fixed this error by running:
`sudo dnf install python3-ansible-runner`